provider/azure/internal/imageutils/images.go

```diff
@@ -152,6 +166,17 @@ func ubuntuSKU(
 	var v1SKU string
 	for _, img := range result.VirtualMachineImageResourceArray {
 		skuName := *img.Name
+		if arch == corearch.ARM64 {
```

Comment:
Can we tweak this so that we switch on skuName at the top level and then in each case block check arch, preferGen1Image conditions  etc.  It will be cleaner.